spec/unit/functions/new_spec.rb

```diff
@@ -28,7 +28,7 @@
       $x = Integer.new(undef)
       notify { "one${x}word": }
     MANIFEST
-    )}.to raise_error(Puppet::Error, /expects an Integer value, got Undef/)
+    )}.to raise_error(Puppet::Error, /cannot be converted to Integer/)
```

Comment:
should check that undef is mentioned in the error

Reply:
this as opposed to getting an empty string in the output or no clue at all what was wrong.

Reply:
The full message says "A Value of type Undef cannot be converted to Integer".